.github/workflows/notification-worker-lambdas.yml

```diff
@@ -0,0 +1,105 @@
+name: Notification worker lambdas
+
+on:
+  push:
+    branches:
+      - "**"
+    paths:
+      - notificationworkerlambda/**
+      - common/**
+      - commoneventconsumer/**
+      - cdk/**
+      - project/**
+      - build.sbt
+      - .github/workflows/notification-worker-lambdas.yml
+  workflow_dispatch:
+
+# allow queued workflows to interrupt previous runs
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: corretto
+          cache: sbt
+          java-version: 11
+
+      - uses: actions/setup-node@v3
+        with:
+          cache: npm
+          cache-dependency-path: cdk/package-lock.json
+          node-version-file: .nvmrc
+
+      - uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: eu-west-1
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+
+      - name: Login to ECR
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          mask-password: 'true'
+
+      - name: Compile and test
+        run: sbt "project notificationworkerlambda" "compile" "test"
+
+      - name: Build docker image
+        run: sbt docker:publishLocal
+
+      - name: Publish docker image
+        run: |
+          docker tag notificationworkerlambda:DEV $NOTIFICATION_LAMBDA_REPOSITORY_URL:$BUILD_NUMBER
+          docker push ${NOTIFICATION_LAMBDA_REPOSITORY_URL}:${BUILD_NUMBER}
+        env:
+          NOTIFICATION_LAMBDA_REPOSITORY_ID: ${{ secrets.NOTIFICATION_LAMBDA_REPOSITORY_ID }}
+          NOTIFICATION_LAMBDA_REPOSITORY_URL: ${{ secrets.NOTIFICATION_LAMBDA_REPOSITORY_URL }}
+          BUILD_NUMBER: ${{ github.run_number }}
+
+      - name: CDK synth
+        run: |
+          cd notificationworkerlambda/cdk
+          yarn install
+          yarn test
+          yarn synth
+          cd ../../cdk
+          npm install
+          npm run lint
+          npm test
+          npm run synth
```

Comment:
just a stylistic thing but I'd probably split this into 2 separate steps with different names, and possibly use the working-directory parameter, but only if you want to.
work directory parameter I feel is slightly more explicit and intuitive, but just my opinion.
Splitting into 2 distinct steps will be more intuitive to future developers debugging an issue, where exactly this has failed. And you never know when Github might add a new feature to show you how long a step has run for historically, to suggest how long might be left, so this might be handy to have split, but neither of these are amazing reasons to do this if you disagree

Reply:
Thanks, I've split these into 2 separate steps.